File: scbp.api-gateway/src/main/java/cn/halower/ZuulConfig.java
package cn.halower;

import org.springframework.cloud.netflix.zuul.filters.discovery.PatternServiceRouteMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ZuulConfig {
        @Bean
        public PatternServiceRouteMapper serviceRouteMapper() {
            return new PatternServiceRouteMapper(
                    "scbp.api-([^-]+)-service",
                    "$1"){
                @Override
                public String apply(final String serviceId) {
                    String route =  super.apply(serviceId);
                    System.out.println(serviceId + " -> " +route);
                    return route;
                }
            };
        }
}

File: scbp-gateway-server/src/main/kotlin/com/halower/scbp/apigateway/config/ZuulConfig.kt
package com.halower.scbp.apigateway.config

import org.slf4j.LoggerFactory
import org.springframework.cloud.netflix.zuul.filters.discovery.PatternServiceRouteMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ZuulConfig {
    private val log = LoggerFactory.getLogger(ZuulConfig::class.java)
    @Bean
    fun serviceRouteMapper(): PatternServiceRouteMapper {
        return object : PatternServiceRouteMapper(
                "([^-]+)-service",
                "$1") {
            override fun apply(serviceId: String): String {
                val route = super.apply(serviceId)
                log.info("$serviceId -> $route")
                return route
            }
        }
    }
}
